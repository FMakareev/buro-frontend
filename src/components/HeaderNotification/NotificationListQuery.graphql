query HeaderNotificationList($clientid: String, $bankid: String, $status:String){
  notificationlist(clientid: $clientid, bankid: $bankid,status:$status){
    id
    status
    date
    readed
    client {
      id
      email
      role
    }
    bank {
      id
      bankName
      email
      role
    }
    bureau {
      id

    }
  }
}
