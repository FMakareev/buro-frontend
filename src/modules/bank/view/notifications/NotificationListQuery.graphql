query($bankid: String){
  notificationlist(bankid: $bankid){
      id
      status
      date
      readed
      client{
        id
        firstName
        lastName
        patronymic
      }
  }
}
