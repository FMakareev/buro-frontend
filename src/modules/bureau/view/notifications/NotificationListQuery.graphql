query($bureau: String){
  notificationlist(bureau: $bureau){
      id
      status
      date
      message
      client{
        id
        firstName
        lastName
        patronymic
      }
  }
}
