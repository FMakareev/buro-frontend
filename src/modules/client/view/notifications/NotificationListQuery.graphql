query($clientid: String){
  notificationlist(clientid: $clientid) {
    id
    status
    date
    bank {
      id
      firstName
      lastName
      patronymic
    }
  }
}
