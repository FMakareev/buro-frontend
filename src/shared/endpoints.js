export const GRAPHQL = '/graphql';
export const EXCEL_UPLOAD = '/upload/excel';
export const EXCEL_DOWNLOAD = '/download/excel';
export const USER_AUTH = '/user/auth';
export const USER_LOGOUT = '/user/logout';
