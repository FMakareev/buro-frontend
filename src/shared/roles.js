export const ROLE_CLIENT = 'client';
export const ROLE_BANK = 'bank';
export const ROLE_BUREAU= 'bureau';
