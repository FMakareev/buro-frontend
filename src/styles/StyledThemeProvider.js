import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { TextVariant } from './variants/textVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#093971',
  color2: '#EFF9FC',
  color3: '#E7EDF8',
  color4: '#61C3E2',
  color5: '#A2A2A2',
  color6: '#0F60BD',
  color7: '#072C57',
  color8: '#61C3E2',
  color9: '#EB5757',
  color10: '#E7EDF8',
  color11: '',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['576px', '768px', '992px', '1200px'];

const boxShadow = ['4px 8px 16px rgba(28, 65, 84, 0.08)'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        text: TextVariant,
      },
      fontFamily: {
        regular: 'Montserrat Regular',
        medium: 'Montserrat Medium',
        bold: 'Montserrat Bold',
      },
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
