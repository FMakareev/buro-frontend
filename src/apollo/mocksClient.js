import faker from 'faker';
import setupClient from './helpers/apolloClientMock';
import schema from './schema.graphqls';
import { userDocumentList } from './graphql/query/userDocumentList';
import { userList } from './graphql/query/userList';
import { userItem } from './graphql/query/userItem';
import { notificationList } from './graphql/query/notificationList';
import { ROLE_BANK, ROLE_BUREAU, ROLE_CLIENT } from '../shared/roles';

const defaultMocks = {
  Query: () => ({
    userList,
    userItem,
    userEmailItem: (query, { email }) => {
      console.log(query, email);
      switch (email) {
        case 'client@test.com': {
          return {
            ...userItem(),
            email: 'client@test.com',
            role: ROLE_CLIENT,
          };
        }
        case 'bank@test.com': {
          return {
            ...userItem(),
            email: 'bank@test.com',
            role: ROLE_BANK,
          };
        }
        case 'bureau@test.com': {
          return {
            ...userItem(),
            email: 'bureau@test.com',
            role: ROLE_BUREAU,
          };
        }
        default: {
          throw Error(
            JSON.stringify({
              userEmailItem: null,
              errors: [
                {
                  message: 'GraphQL error: user not found',
                },
              ],
            }),
          );
        }
      }
    },
    userDocumentList,
    notificationList,
  }),
  Mutation: () => ({
    /**
     * @params {func} mutation - этот же запрос
     * @params {object} props - аргументы которые были переданы
     * */
    createUser: (mutation, props) => props,
    updateUser: (mutation, props) => props,
    userResetPassword: (mutation, props) => props,
    createNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { createNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    updateNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { updateNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    userPasswordRecovery: (mutation, props) =>
      // throw Error(JSON.stringify({
      //   userPasswordRecovery: null,
      //   errors: [
      //     {
      //       message:'Error!'
      //     }
      //   ]
      // }))
      props,
  }),
};

export const mocksClient = setupClient(defaultMocks, schema)();

export default mocksClient;
