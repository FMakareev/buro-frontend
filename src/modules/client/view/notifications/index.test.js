import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';
import { MockedProvider } from 'react-apollo/test-utils';

import wait from 'waait';
import faker from 'faker';
import { MemoryRouter, Redirect } from 'react-router-dom';
import { Provider as ProviderRedux } from 'react-redux';
import configureStore from 'redux-mock-store';
import { ROLE_BANK } from '@lib/shared/roles';
import mocksClient from '../../../../apollo/mocksClient';

import { ClientsNotificationsPage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import NotificationListQuery from './NotificationListQuery.graphql';

const mockStore = configureStore();

test('ClientsNotificationsPage: пользователь не авторизован', () => {
  const store = mockStore({});

  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={store}>
        <MemoryRouter>
          <ApolloProvider client={mocksClient}>
            <ClientsNotificationsPage />
          </ApolloProvider>
        </MemoryRouter>
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  expect(output.root.findByType(Redirect).props.to).toBe('/logout');
});

test('ClientsNotificationsPage: загрузка пользовательских данных', () => {
  const store = mockStore({
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: ROLE_BANK,
      id: faker.random.uuid(),
    },
  });

  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={store}>
        <MemoryRouter>
          <ApolloProvider client={mocksClient}>
            <ClientsNotificationsPage />
          </ApolloProvider>
        </MemoryRouter>
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});

test('ClientsNotificationsPage: ошибка во время загрузки пользовательских данных', async () => {
  const initialValue = {
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: ROLE_BANK,
      id: faker.random.uuid(),
    },
  };
  const store = mockStore(initialValue);
  const dogMock = {
    request: {
      query: NotificationListQuery,
      variables: {
        id: initialValue.user.id,
      },
    },
    result: {
      errors: [{ message: 'Error!' }],
    },
  };
  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={store}>
        <MemoryRouter>
          <MockedProvider mocks={[dogMock]} addTypename>
            <ClientsNotificationsPage />
          </MockedProvider>
        </MemoryRouter>
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  await wait(0); // небольшая задержка

  const tree = output.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ClientsNotificationsPage: загруженный список нотификаций', async () => {
  const store = mockStore({
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: ROLE_BANK,
      id: faker.random.uuid(),
    },
  });

  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={store}>
        <MemoryRouter>
          <ApolloProvider client={mocksClient}>
            <ClientsNotificationsPage />
          </ApolloProvider>
        </MemoryRouter>
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  await wait(5); // небольшая задержка

  expect(output).toMatchSnapshot();
});
