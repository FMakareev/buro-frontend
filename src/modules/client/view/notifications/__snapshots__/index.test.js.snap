// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientsNotificationsPage: загруженный список нотификаций 1`] = `
<div
  className="sc-bxivhb dUiVDW"
/>
`;

exports[`ClientsNotificationsPage: загрузка пользовательских данных 1`] = `
<div
  className="sc-bxivhb dUiVDW"
/>
`;

exports[`ClientsNotificationsPage: ошибка во время загрузки пользовательских данных 1`] = `
<div
  className="sc-bxivhb dUiVDW"
/>
`;
